### PR TITLE
Add email based auth with PKCE flow

### PR DESCRIPTION
**Pull request by silentworks**
## What kind of change does this PR introduce?

docs update

## What is the current behavior?

No documentation on getting the pkce auth flow working when magic link is opened on a different device.

## What is the new behavior?

Add documentation on getting the pkce auth flow working when magic link is opened on a different device.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-helpers.mdx [hunk 1]
```diff
@@ -2,31 +2,15 @@ import Layout from '~/layouts/DefaultGuideLayout'
...
   id: 'index',
-  title: 'Auth Helpers Overview',
-  description: 'A collection of framework-specific Auth utilities for working with Supabase.',
+  title: 'Server-Side Auth Overview',
+  description: 'Server-Side Auth guides and utilities for working with Supabase.',
   sidebar_label: 'Overview',
 }
 
-A collection of framework-specific Auth utilities for working with Supabase.
+Working with server-side frameworks is a bit different to client-side frameworks, in this section we cover the various ways of handling authentication on the server-side and provide some helper libraries to make the process more seamless when working with Supabase.
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/auth-helpers.mdx:10]:
```suggestion
Working with server-side frameworks is slightly different to client-side frameworks. In this section we cover the various ways of handling server-side authentication and demonstrate how to use the Supabase helper-libraries to make the process more seamless.
```

File: apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx [hunk 2]
```diff
@@ -0,0 +1,236 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+
+export const meta = {
+  title: 'Email Auth with PKCE flow for SSR',
+  description:
+    'Learn how to configure email authentication in your server-side rendering (SSR) application to work with the PKCE flow.',
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx:6]:
```suggestion
  description:
    'Learn how to configure email authentication in your server-side rendering (SSR) application to work with the PKCE flow.',
  subtitle:
    'Learn how to configure email authentication in your server-side rendering (SSR) application to work with the PKCE flow.',
```

File: apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+The Auth Helpers will assist you in implementing user authentication within your server-side rendering (SSR) framework.
+
+<Tabs scrollable size="small" type="underlined" defaultActiveId="nextjs">
+  <TabPanel id="nextjs" label="NextJS">
+    ```bash npm install @supabase/auth-helpers-nextjs @supabase/supabase-js ```
+  </TabPanel>
+  <TabPanel id="sveltekit" label="SvelteKit">
+    ```bash npm install @supabase/auth-helpers-sveltekit @supabase/supabase-js ```
+  </TabPanel>
+</Tabs>
```
Comment (kiwicopple) [inline apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx:20]: just need to double-check that the formatting here is good - I thought the code blocks need to have no indentation (but I could be wrong)

Comment (silentworks) [reply apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx:22]: When testing this locally it seemed to work with the indentation.

Comment (silentworks) [reply apps/docs/pages/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr.mdx:22]: Oh wait it looks like prettier botched that code snippet.

**Review by kiwicopple — APPROVED** [review]

**Pull request merged**